```log
java.lang.RuntimeException: Failed to find the generated JsonAdapter class for class io.goooler.demoapp.common.JsonUtilTest$Repo
	at com.squareup.moshi.internal.Util.generatedAdapter(Util.java:590)
	at com.squareup.moshi.StandardJsonAdapters$1.create(StandardJsonAdapters.java:61)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:146)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:106)
	at com.squareup.moshi.Moshi.adapter(Moshi.java:80)
	at io.goooler.demoapp.common.JsonUtilTest.JsonUtil fromJson(String)(JsonUtilTest.kt:95)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.ClassNotFoundException: io.goooler.demoapp.common.JsonUtilTest_RepoJsonAdapter
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at com.squareup.moshi.internal.Util.generatedAdapter(Util.java:564)
	... 48 more

```

--- a/buildSrc/src/main/kotlin/Extensions.kt
+++ b/buildSrc/src/main/kotlin/Extensions.kt
@@ -57,6 +57,9 @@ fun DependencyHandler.androidTestImplementations(vararg names: Any): Array<Depen
 fun DependencyHandler.testImplementations(vararg names: Any): Array<Dependency?> =
   config("testImplementation", *names)
 
+fun DependencyHandler.kaptTests(vararg names: Any): Array<Dependency?> =
+  config("kaptTest", *names)
+
 fun PluginAware.applyPlugins(vararg names: String) {
   apply { names.forEach(::plugin) }
 }

--- a/common/build.gradle.kts
+++ b/common/build.gradle.kts
@@ -29,4 +29,6 @@ dependencies {
   implementations(*Libs.coil)
   debugImplementations(Libs.chuckerDebug)
   releaseImplementations(Libs.chuckerRelease)
+
+  kaptTests(Libs.moshiCompiler)
 }

--- a/common/src/main/kotlin/io/goooler/demoapp/common/util/JsonUtil.kt
+++ b/common/src/main/kotlin/io/goooler/demoapp/common/util/JsonUtil.kt
@@ -4,12 +4,13 @@ package io.goooler.demoapp.common.util
 
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.Types
+import org.intellij.lang.annotations.Language
 
 object JsonUtil {
   @PublishedApi
   internal val moshi: Moshi = Moshi.Builder().build()
 
-  inline fun <reified T> fromJson(string: String): T? = try {
+  inline fun <reified T> fromJson(@Language("JSON") string: String): T? = try {
     moshi.adapter(T::class.java).fromJson(string)
   } catch (e: Exception) {
     e.printStackTrace()
@@ -17,7 +18,7 @@ object JsonUtil {
   }
 
   fun <T> fromJson(
-    string: String,
+    @Language("JSON") string: String,
     rawType: Class<*>,
     vararg typeArguments: Class<*>
   ): T? = try {
@@ -27,6 +28,7 @@ object JsonUtil {
     null
   }
 
+  @Language("JSON")
   inline fun <reified T> toJson(value: T?): String? = try {
     moshi.adapter(T::class.java).toJson(value)
   } catch (e: Exception) {
@@ -40,4 +42,5 @@ inline fun <reified T> String.fromJson(): T? = JsonUtil.fromJson(this, T::class.
 inline fun <reified T> String.fromJson(rawType: Class<*>, vararg typeArguments: Class<*>): T? =
   JsonUtil.fromJson(this, rawType, *typeArguments)
 
+@Language("JSON")
 fun Any?.toJson(): String? = JsonUtil.toJson(this)

--- a/common/src/main/kotlin/io/goooler/demoapp/common/util/JsonUtil.kt
+++ b/common/src/main/kotlin/io/goooler/demoapp/common/util/JsonUtil.kt
@@ -37,7 +37,7 @@ object JsonUtil {
   }
 }
 
-inline fun <reified T> String.fromJson(): T? = JsonUtil.fromJson(this, T::class.java)
+inline fun <reified T> String.fromJson(): T? = JsonUtil.fromJson(this)
 
 inline fun <reified T> String.fromJson(rawType: Class<*>, vararg typeArguments: Class<*>): T? =
   JsonUtil.fromJson(this, rawType, *typeArguments)

--- a/common/src/test/kotlin/io/goooler/demoapp/common/JsonUtilTest.kt
+++ b/common/src/test/kotlin/io/goooler/demoapp/common/JsonUtilTest.kt
@@ -4,6 +4,7 @@ import com.squareup.moshi.JsonClass
 import io.goooler.demoapp.common.util.JsonUtil
 import io.goooler.demoapp.common.util.fromJson
 import io.goooler.demoapp.common.util.toJson
+import org.intellij.lang.annotations.Language
 import org.junit.Assert.assertTrue
 import org.junit.Test
 
@@ -63,6 +64,7 @@ class JsonUtilTest {
   }
 
   companion object {
+    @Language("JSON")
     private val firstStr = """
       {
         "id": 126987864,
@@ -73,6 +75,7 @@ class JsonUtilTest {
       }
     """.trimIndent()
 
+    @Language("JSON")
     private val secondStr = """
       {
         "id": 374913489,
@@ -84,6 +87,7 @@ class JsonUtilTest {
     """.trimIndent()
 
     // https://api.github.com/users/goooler/repos?&page=1&per_page=2
+    @Language("JSON")
     private val strArray = "[$firstStr,$secondStr]"
 
     private val firstBean = Repo(126987864, "1024_hosts", Repo.Owner("Goooler"))

--- a/common/src/test/kotlin/io/goooler/demoapp/common/JsonUtilTest.kt
+++ b/common/src/test/kotlin/io/goooler/demoapp/common/JsonUtilTest.kt
@@ -17,7 +17,7 @@ class JsonUtilTest {
 
   @Test
   fun `JsonUtil fromJson(String, Class, Class)`() {
-    val array: Array<Repo> = JsonUtil.fromJson(strArray, Array::class.java, Repo::class.java)
+    val array: List<Repo> = JsonUtil.fromJson(strArray, List::class.java, Repo::class.java)
       ?: throw Exception("Parse json error")
     assertTrue(array.first() == firstBean)
     assertTrue(array[1] == secondBean)
@@ -37,7 +37,7 @@ class JsonUtilTest {
 
   @Test
   fun `String fromJson(Class, Class)`() {
-    val array: Array<Repo> = strArray.fromJson(Array::class.java, Repo::class.java)
+    val array: List<Repo> = strArray.fromJson(List::class.java, Repo::class.java)
       ?: throw Exception("Parse json error")
     assertTrue(array.first() == firstBean)
     assertTrue(array[1] == secondBean)
@@ -66,24 +66,12 @@ class JsonUtilTest {
   companion object {
     @Language("JSON")
     private val firstStr = """
-      {
-        "id": 126987864,
-        "name": "1024_hosts",
-        "owner": {
-          "login": "Goooler"
-        }
-      }
+      {"id":126987864,"name":"1024_hosts","owner":{"login":"Goooler"}}
     """.trimIndent()
 
     @Language("JSON")
     private val secondStr = """
-      {
-        "id": 374913489,
-        "name": "AndroidUiDemo",
-        "owner": {
-          "login": "Goooler"
-        }
-      }
+      {"id":374913489,"name":"AndroidUiDemo","owner":{"login":"Goooler"}}
     """.trimIndent()
 
     // https://api.github.com/users/goooler/repos?&page=1&per_page=2

